src/plugins/node-workspace.ts

```diff
@@ -52,39 +55,39 @@ export default class NodeWorkspaceDependencyUpdates extends ManifestPlugin {
   // those that did not update (no user facing commits).
   private async getAllWorkspacePackages(
     rpUpdatedPkgs: PathPkgJson
-  ): Promise<Map<string, Package>> {
+  ): Promise<Map<string, [Package, string]>> {
     const nodePkgs = new Map();
     for (const pkg of this.config.parsedPackages) {
       if (pkg.releaseType !== 'node' || pkg.path === '.') {
         continue;
       }
       const path = `${pkg.path}/package.json`;
-      let contents: PackageJson;
+      let contents: [PackageJson, string];
       const alreadyUpdated = rpUpdatedPkgs.get(path);
       if (alreadyUpdated) {
         contents = alreadyUpdated;
       } else {
-        const fileContents = await this.gh.getFileContents(path);
-        contents = JSON.parse(fileContents.parsedContent);
+        const {parsedContent} = await this.gh.getFileContents(path);
+        contents = [JSON.parse(parsedContent), parsedContent];
       }
       this.log(
         `loaded ${path} from ${alreadyUpdated ? 'existing changes' : 'github'}`,
         CheckpointType.Success
       );
-      nodePkgs.set(path, new Package(contents, path));
+      nodePkgs.set(path, [new Package(contents[0], path), contents[1]]);
     }
     return nodePkgs;
   }
 
   private async runLernaVersion(
     rpUpdatedPkgs: PathPkgJson,
-    allPkgs: Map<string, Package>
-  ): Promise<Map<string, Package>> {
+    allPkgs: Map<string, [Package, string]>
+  ): Promise<Map<string, [Package, string]>> {
```

Comment:
I'm a little worried about the implicit behavior we've introduced here, where someone needs to know that the first argument is the package object, and the second argument is the raw string.

I think I'd rather that we create a new interface that included the Package type and the raw content type.

@joeldodge79 does this sound reasonable to you?

Reply:
We can give a name to the elements of the tuple like `[package: Package, rawContent: string]`?

Reply:
The line that concerns me the most is this:

`allUpdated.map(p => [p.location, [p, allPkgs.get(p.location)![1][1]]])`

Because you need to know that `1` represents the original content.

It would be nicer to be able to do something like:

```js
allUpdated.map(p => [p.location, [p, allPkgs.get(p.location)![1][1]]])
```

With the triple nested arrays, I find it a bit hard to know what's going on, I think named properties could help clean this up a bit.

Would be curious to know @joeldodge79's feedback too, who implemented most of this class.

Reply:
I agree that [L207](https://github.com/googleapis/release-please/pull/923/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts#diff-811e4913774d18713311ccab03d320c30f1a04286989067b28003404fd88197dR207) is hard on my brain. Does TS supported namedtuples?

What if we made an assumption that all files under the control of a plugin had the same indentation style? Then we could set a `indentContent` instance property on the plugin using the first raw content we come across and reference that for every call rather than passing around the raw content for each file. Alternatively we could make `jsonStringify` a simple class that takes `indentedContent: string` as a constructor and initialize it once for the plugin.

But if we really want the flexibility of supporting different indentation styles per plugin-scope in a monorepo then this seems like the only solution.

Reply:
> What if we made an assumption that all files under the control of a plugin had the same indentation style?

If we think this is a reasonable assumption, I like this idea as it cuts down on the amount of information we need to pass around.